src/Compilers/CSharp/Portable/Symbols/Source/SourceLocalSymbol.cs

```diff
@@ -395,7 +395,7 @@ internal void SetTypeWithAnnotations(TypeWithAnnotations newType)
 
             Debug.Assert((object)originalType == null ||
                 originalType.IsErrorType() && newType.Type.IsErrorType() ||
-                TypeSymbol.Equals(originalType, newType.Type, TypeCompareKind.ConsiderEverything2));
+                _type.Value.TypeSymbolEquals(newType, TypeCompareKind.ConsiderEverything));
```

Comment:
>_type.Value [](start = 16, length = 11)

I think we should use ```originalType``` here only we should adjust its declaration as follows:
```
TypeWithAnnotations? originalType = _type?.Value;
```

and adjust null check as appropriate.